### ISSUE TITLE
Support equivalent of IQueryable<ComposablePartDefinition> Parts in System.Composition

### ISSUE THREAD

**Issue by tedescomicchioutlook** [dotnet/corefx#26507]
`System.ComponentModel.Composition` (MEF1) has the ability to list available `Parts` through `System.ComponentModel.Composition.Primitives`. It would be useful to have this ability in `System.Composition` (MEF2).

**Comment by GFlisch** [comment]
I agree with you, this feature is important when we want to query implementation of an interface for sample and offer a way to choose the implementation (via a gui for sample).
The ConfigurationContainer contains an internal method where the info is there, can you do this public?
